templates/pull_request_template.md

```diff
@@ -0,0 +1,20 @@
+All Submissions:
+
+* [ ] Have you followed the guidelines in our Contributing document?
+* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
+
+<!-- You can erase any parts of this template not applicable to your Pull Request. -->
+### Checklist before requesting a review
+- [ ] I have performed a self-review of my code
+- [ ] If it is a core feature, I have added thorough tests.
+### New Feature Submissions:
+
+1. [ ] Does your submission pass tests?
+2. [ ] Have you lint your code locally before submission?
```

Comment:
```suggestion
2. [ ] Did you install pre-commit hooks before committing?
```

Reply:
New features should come with a problem statement and why this proposal / MR solves it. We dont just want to check the code but validate the idea. In theory it would make more sense to create a proposal issue then create the MR attached to it.

Reply:
Yeah I agree with the approach of create issue first before solving it. Reason being is that might find that there is a reason we have not gone that route before spending time on creating the MR.  👍

Reply:
Yep agree Worm 👍  @wenkanglu was going to look at adding a matching ISSUE_TEMPLATE.md too. So I think we can have a checkbox for "I have created an issue for this proposal, if it is non-trivial" here (or similar)